lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -687,9 +687,23 @@ def generate_clone_spec(src_config)
         # We should get here with the customizations set, either by a plugin or a --cspec
         fatal_exit 'Windows clones need a customization identity. Try passing a --cspec or making a --cplugin' if cust_spec.identity.props.empty?
 
-        identification = RbVmomi::VIM.CustomizationIdentification(
-          joinWorkgroup: cust_spec.identity.identification.joinWorkgroup
-        )
+        # If --cdomain matches what is in --cspec then use identification from the --cspec, else use --cdomain
+        domain = get_config(:customization_domain)
+        if domain
+          if domain == cust_spec.identity.identification.joinDomain
```

Comment:
BTW what this is saying in a confusing way is that since each branch of the `if` statement assigns the same variable, that instead of

```
if a
  foo = 'bar'
else
  foo = 'baz'
end
```

you can say

```
foo = if a
            'foo'
          else
            'bar'
          end
```
And if you end up moving this all to a private method you can skip the assignment since the return value of a method is the result of the last expression.